src/plugins/condition/components/Criterion.vue

```diff
@@ -0,0 +1,145 @@
+                        :value="telemetryOption.identifier"
+                >
+                    {{ telemetryOption.name }}
+                </option>
+            </select>
+        </span>
+        <span class="controls">
+            <select v-model="selectedFieldOption"
+                    @change="persist"
+            >
+                <option value="">- Select Field -</option>
+                <option v-for="option in telemetryMetadata"
+                        :key="option.key"
+                        :value="option"
+                >
+                    {{ option.name }}
+                </option>
+            </select>
+        </span>
+        <span class="controls">
+            <select v-model="selectedOperationOption"
+                    @change="updateOperationInput"
+            >
+                <option value="">- Select Comparison -</option>
+                <option v-for="option in operations"
+                        :key="option.name"
+                        :value="option.name"
+                >
+                    {{ option.text }}
+                </option>
+            </select>
+            <input v-if="isInputOperation"
+                   v-model="comparisonInputValue"
+                   class="t-condition-name-input"
+                   type="text"
+                   @change="persist"
+            >
+        </span>
+    </span>
+</li>
+
+</template>
+
+<script>
+import { OPERATIONS } from '../utils/operations';
+
+export default {
+    inject: ['openmct'],
+    props: {
+        criterion: {
+            type: Object,
+            required: true
+        },
+        telemetry: {
+            type: Array,
+            required: true,
+            default: () => []
+        },
+        index: {
+            type: Number,
+            required: true
+        },
+        condition: {
+            type: Object,
+            required: true
+        },
+        trigger: {
+            type: String,
+            required: true
+        }
+    },
+    data() {
+        return {
+            telemetryMetadata: {},
+            operations: OPERATIONS,
+            selectedTelemetryObject: '',
+            selectedFieldOption: {},
+            selectedOperationOption: '',
+            operationValue: '',
+            comparisonInputValue: '',
+            isInputOperation: false,
+            rowLabel: ''
+        }
+    },
+    computed: {
+        setRowLabel: function () {
+            let operator = this.trigger === 'all' ? 'and ': 'or ';
+
+            return (this.index !== 0 ? operator : '') + 'when';
+        }
+    },
+    mounted() {
+        this.initialize();
+    },
+    updated() {
+        this.persist();
+    },
+    methods: {
+        initialize() {
+            this.selectedTelemetryObject = this.criterion.telemetry;
+            this.selectedFieldOption = this.criterion.metadata;
+            this.selectedOperationOption = this.criterion.operation;
```

Comment:
Do we need to assign this.criterion.xxx to this.selectedxxx? Would it be possible to use this.criterion.xxx directly in the markup?

Reply:
done